src/Corax/Queries/UnaryMatch.Ordinal.cs

```diff
@@ -132,20 +129,17 @@ private static int FillFuncNumerical<TComparer>(ref UnaryMatch<TInner, TValueTyp
                             if (read)
                                 isMatch = comparer.Compare((long)(object)currentType, resultX);
                         }
-                        else if (type is IndexEntryFieldType.List)
+                        else if (type is IndexEntryFieldType.List or IndexEntryFieldType.TupleList)
                         {
                             var iterator = reader.ReadMany(match._fieldId);
-                            var unacceptableItem = false;
                             while (iterator.ReadNext())
                             {
-                                if (comparer.Compare((long)(object)currentType, iterator.Long) == false)
+                                if (comparer.Compare((long)(object)currentType, iterator.Long))
                                 {
-                                    unacceptableItem = true;
+                                    isMatch = true;
                                     break;
                                 }
                             }
-
-                            isMatch = unacceptableItem == false;
                         }
```

Comment:
The main reason behind those changes is that when we query `Data[] < X` we want to check if at least one term is acceptable in the list, not every item inside the list.